src/submission/validation-clinical/utils.ts

```diff
@@ -62,45 +56,167 @@ export const buildSubmissionUpdate = (
+      if (clinicalObject[fieldName] !== record[fieldName]) {
+        submissionUpdates.push(buildSubmissionUpdate(record, clinicalObject[fieldName], fieldName));
+      }
+    }
+  }
+  return submissionUpdates;
 };
+
+export const buildClinicalValidationResult = (results: RecordValidationResult[]) => {
+  const stats = {
+    [ModificationType.NEW]: [] as number[],
+    [ModificationType.NOUPDATE]: [] as number[],
+    [ModificationType.UPDATED]: [] as number[],
+    [ModificationType.ERRORSFOUND]: [] as number[],
+  };
+  let dataErrors: SubmissionValidationError[] = [];
+  let dataUpdates: SubmissionValidationUpdate[] = [];
+
+  results.forEach(result => {
+    stats[result.type].push(result.index);
+    if (result.type === ModificationType.UPDATED) {
+      dataUpdates = dataUpdates.concat(result.resultArray as SubmissionValidationUpdate[]);
+    } else if (result.type === ModificationType.ERRORSFOUND) {
+      dataErrors = dataErrors.concat(result.resultArray as SubmissionValidationError[]);
+    }
+  });
+
+  return {
+    stats: stats,
+    dataErrors: dataErrors,
+    dataUpdates: dataUpdates,
+  };
+};
+
+export const buildMultipleRecordValidationResults = (
+  records: ReadonlyArray<SubmittedClinicalRecord>,
+  commonErrorProperties: {
+    type: DataValidationErrors;
+    fieldName: FieldsEnum | ClinicalInfoFieldsEnum;
+    info?: any;
+  },
+): RecordValidationResult[] => {
+  const validationResults = records.map(record => {
+    return buildRecordValidationResult(
+      record,
+      buildSubmissionError(
+        record,
+        commonErrorProperties.type,
+        commonErrorProperties.fieldName,
+        commonErrorProperties.info,
+      ),
+      {},
+    );
+  });
+
+  return validationResults;
+};
+
+export namespace ClinicalSubmissionRecordsOperations {
+  // this function will mutate a DonorRecordsOrganizer
+  export function addRecord(
+    type: ClinicalEntityType,
+    records: SubmittedClinicalRecordsMap,
+    record: SubmittedClinicalRecord,
+  ) {
+    switch (type) {
+      case ClinicalEntityType.PRIMARY_DIAGNOSIS:
+      case ClinicalEntityType.DONOR: {
+        records[type] = record;
+        break;
+      }
+      case ClinicalEntityType.SPECIMEN: {
+        if (!records[type]) {
+          records[type] = [];
+        }
+        (records[type] as SubmittedClinicalRecord[]).push(record as SubmittedClinicalRecord);
+        break;
+      }
+      default:
+        throw new Error(`Can't add record with type: ${type}`);
+    }
+  }
+
+  export function getRecordsAsArray(
+    type: ClinicalEntityType,
+    records: DeepReadonly<SubmittedClinicalRecordsMap>,
+  ): ReadonlyArray<SubmittedClinicalRecord> {
+    const recordsOfInterest = records[type];
+    if (!recordsOfInterest) {
+      return [];
+    } else if (Array.isArray(recordsOfInterest)) {
+      return recordsOfInterest;
+    } else {
+      return [recordsOfInterest as SubmittedClinicalRecord];
+    }
+  }
+
+  export function getDonorRecord(
```

Comment:
I agree it makes sense to have one function to get record but then we have to keep doing `as SubmittedClinicalRecord` or `as SubmittedClinicalRecord[]` everytime we try to access from submittedRecord...